fe/fe-core/src/main/java/com/starrocks/authentication/AuthenticationManager.java

```diff
@@ -157,4 +168,93 @@ private void updateUserNoLock(
+     */
+    public void save(DataOutputStream dos) throws IOException {
+        try {
+            // 1 json for myself,1 json for number of users, 2 json for each user(kv)
+            final int cnt = 1 + 1 + userToAuthenticationInfo.size() * 2;
+            SRMetaBlockWriter writer = new SRMetaBlockWriter(dos, AuthenticationManager.class.getName(), cnt);
+            // 1 json for myself
+            writer.writeJson(this);
+            // 1 json for num user
+            writer.writeJson(userToAuthenticationInfo.size());
+            Iterator<Map.Entry<UserIdentity, UserAuthenticationInfo>> iterator =
+                    userToAuthenticationInfo.entrySet().iterator();
+            while (iterator.hasNext()) {
+                // 2 json for each user(kv)
+                Map.Entry<UserIdentity, UserAuthenticationInfo> entry = iterator.next();
+                writer.writeJson(entry.getKey());
+                writer.writeJson(entry.getValue());
+            }
+            writer.close();
+        } catch (SRMetaBlockException e) {
+            IOException exception = new IOException("failed to save AuthenticationManager!");
+            exception.initCause(e);
+            throw exception;
+        }
+    }
+
+    public static AuthenticationManager load(DataInputStream dis) throws IOException, DdlException {
+        try {
+            SRMetaBlockReader reader = new SRMetaBlockReader(dis, AuthenticationManager.class.getName());
+            AuthenticationManager ret = null;
+            try {
+                // 1 json for myself
+                ret = (AuthenticationManager) reader.readJson(AuthenticationManager.class);
+                ret.userToAuthenticationInfo = new HashMap<>();
+                // 1 json for num user
+                int numUser = (int) reader.readJson(int.class);
+                LOG.info("loading {} users", numUser);
+                for (int i = 0; i != numUser; ++i) {
+                    // 2 json for each user(kv)
+                    UserIdentity userIdentity = (UserIdentity) reader.readJson(UserIdentity.class);
+                    UserAuthenticationInfo userAuthenticationInfo =
+                            (UserAuthenticationInfo) reader.readJson(UserAuthenticationInfo.class);
+                    userAuthenticationInfo.analyze();
+                    ret.userToAuthenticationInfo.put(userIdentity, userAuthenticationInfo);
+                }
+            } catch (SRMetaBlockEOFException eofException) {
+                LOG.warn("got EOF exception, ignore, ", eofException);
+            } finally {
+                reader.close();
+            }
+            assert ret != null; // can't be NULL
+            LOG.info("loaded {} users", ret.userToAuthenticationInfo.size());
+            return ret;
+        } catch (SRMetaBlockException | AuthenticationException e) {
+            DdlException exception = new DdlException("failed to save AuthenticationManager!");
+            exception.initCause(e);
```

Comment:
```suggestion
            DdlException exception = new DdlException("failed to save AuthenticationManager!", e);
```
we can use one line.

Reply:
done